### PR TITLE
Kc 1970 tanstack query refactor pages 6 8

### PR DESCRIPTION
**Pull request by JoseToitio**

### CODE REVIEW

File: src/pages/itos/index.tsx [hunk 1]
```diff
@@ -96,157 +100,63 @@ export const displayITOpacks = (
...
+  const {
+    data,
+    fetchNextPage: nextPageITOs,
+    isFetching,
+    hasNextPage,
+    isLoading: isLoadingITOs,
+    refetch: refetchITOs,
+  } = useInfiniteQuery(
+    'ITOss',
+    async ({ pageParam = 1 }) => {
```
Comment (lucasportella) [inline src/pages/itos/index.tsx:121]: too complex, this should be a separated function.

File: src/pages/itos/index.tsx [hunk 2]
```diff
@@ -96,157 +100,63 @@ export const displayITOpacks = (
...
+      const dataITOs = await requestITOs(router, pageParam);
+      const assets = await requestAssetsList(dataITOs);
+      await parseITOsRequest(dataITOs, assets);
+      return dataITOs;
+    },
+    {
+      getNextPageParam: lastPage => {
```
Comment (lucasportella) [inline src/pages/itos/index.tsx:128]: same here

File: src/pages/itos/index.tsx [hunk 3]
```diff
@@ -96,157 +100,63 @@ export const displayITOpacks = (
...
-    ITO.packData.forEach(async packInfo => {
-      const keyPrecision = await getPrecision(packInfo.key);
-      packInfo.packs.forEach(pack => {
-        pack.price = pack.price / 10 ** keyPrecision;
-        pack.amount = pack.amount / 10 ** assetPrecision;
-      });
-    });
-    return ITO;
-  };
+  const itemsITOs = useMemo(() => {
```
Comment (lucasportella) [inline src/pages/itos/index.tsx:140]: dont use useMemo for iterations below 1k items. Consider reading [here](https://react.dev/learn/you-might-not-need-an-effect#how-to-tell-if-a-calculation-is-expensive)

File: src/services/requests/ito/index.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  } catch (error) {
+    console.error(error);
+  }
+  return;
+};
+
+export const requestAssetsList = async (
+  data: IITOsResponse,
+): Promise<any | void> => {
+  if (data) {
```
Comment (lucasportella) [inline src/services/requests/ito/index.ts:29]: consider checking the length of the data, since it returns an object, will always be true

**Review by lucasportella — CHANGES_REQUESTED** [review]

**Review by lucasportella — APPROVED** [review]

**Review by Doug-Fernando — APPROVED** [review]

**Pull request merged**